src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,368 @@
+
+Within an interop type, you can declare several different types of `external`
+interop members:
+
+- Constructors. When called, these members construct a new JS object whose
+  constructor is defined by the name of the extension type using `new`. For
+  example, calling `Array()` in Dart will generate a JS invocation that looks
+  like `new Array()`. Similarly, calling `Array.withLength(10)` will generate a
+  JS invocation that looks like `new Array(10)`. Note that the JS invocations of
+  all `external` interop constructors follow the same semantics, regardless of
+  whether they're given a name in Dart or if they are a factory.
+
+- `static` members. Like constructors, these members use the name of the
+  extension type to generate the JS code. For example, calling `Array.isArray()`
+  will generate a JS invocation that looks like `Array.isArray()`. Like
+  top-levels, you can declare `static` methods, getters, setters, and fields.
+
+- Instance members. Like with other Dart types, these members require an
+  instance in order to be used. These members get, set, or invoke properties on
+  the instance. For example:
+
+  ```dart
+    final array = Array();
+    array.length = 10;
+    final length = array.length;
+    for (final i = 0; i <= 10; i++) {
+      array[i] = i.toJS;
+      assert(array[i] == array.at(i));
+    }
+  ```
+
+  The call to `array.length` gets the value of the `length` property of `array`.
+  The call to `array.length = 10` sets the value of the `length` property. The
+  call to `array.at(i)` calls the function in the `at` property of `array` with
+  the given arguments and returns the value of that call. Like top-levels and
+  `static` members, you can declare instance methods, getters, setters, and
+  fields.
+
+- Operators. There are only two `external` interop operators allowed in interop
+  types: `[]` and `[]=`. These are instance members that match the semantics of
+  JS' [property accessors]. In the above example, `array[i]` gets the value in
+  the `i`th slot of `array`, and `array[i]` sets the value in that slot to
+  `i.toJS`. We expose other JS operators through [utility functions] in
+  `dart:js_interop`.
+
+In the [JS types] section, you'll see that there exists a predefined JS type for
+`Array` called `JSArray`. Here, we're writing another interop type for the same
+JS type and just *viewing* it differently, so there is no conflict. If we
+wanted, we could have made the representation type `JSArray`:
+
+```dart
+extension type Array._(JSArray<JSAny?> _) implements JSArray<JSAny?> {}
+```
+
+Lastly, like any other extension type, you're allowed to declare any
+non-`external` members in the interop type. `isZeroLength` is one such example.
+
+#### Object literal constructors
+
+It is useful sometimes to create a JS [object literal] that simply contains a
+a number of properties and their values. In order to do this, we allow you to
+use a constructor with only named arguments:
```

Comment:
It may be helpful to provide a rationale for why this syntax. Unlike the other syntaxes we have, the reason a constructor with positional arguments means one thing, and with named arguments means another, it's still very subtle.

It may help to have a short section first about constructors in general, and explain the details there? We can then draw a parallel with JavaScript, highlighting that because functions in JavaScript only take positional arguments, we adopted in Dart that a constructor using only positional arguments is considered a reference to a constructor function.